nbqa/__main__.py

```diff
@@ -270,11 +287,40 @@ def _get_arg(
     if Path(root_dir).is_dir():
         arg = Path(tmpdirname) / Path(root_dir).resolve().relative_to(project_root)
     else:
-        assert len(nb_to_py_mapping) == 1
-        arg = next(iter(nb_to_py_mapping.values()))
+        arg = nb_to_py_mapping[Path(root_dir)]
     return arg
 
 
+def _get_all_args(
+    root_dirs: List[str],
+    tmpdirname: str,
+    nb_to_py_mapping: Dict[Path, Path],
+    project_root: Path,
+) -> Tuple[Path, ...]:
+    """
+    Get all arguments to run command against.
+
+    Parameters
+    ----------
+    root_dirs
+        All notebooks or directories third-party tool is being run against.
+    tmpdirname
+        Temporary directory where converted notebooks are stored.
+    nb_to_py_mapping
+        Mapping between notebooks and Python files corresponding to converted notebooks.
+    project_root
+        Root of repository, where .git / .hg / .nbqa.ini file is.
+
+    Returns
+    -------
+    List[Path]
+        All notebooks or directories to run third-party tool against.
+    """
+    return tuple(
```

Comment:
@MarcoGorelli : In order to use tuple, you have used `tuple(generator expression)` instead of earlier list comprehension. list comprehension performs better than tuple with generator expressions.

Ref: https://stackoverflow.com/questions/43808180/python-3-x-list-comprehension-vs-tuple-generator (refer accepted answer)

Also performance difference between tuple and lists is during creation only (also extra memory allocated for lists). While accessing both produce same performance.
Ref: https://stackoverflow.com/questions/68630/are-tuples-more-efficient-than-lists-in-python (refer accepted answer)

Based on your latest change, looks like you opted for less performant approach.

Reply:
:laughing: oops, thanks for catching me!